### PR TITLE
VZ-8954: Uptake Jaeger 1.42.0

### PR DESCRIPTION
**Pull request by pshirodkar-orcl**
Uptake Jaeger 1.42.0.
Remove jaeger hotrod e2e suites since these are essentially same as jaeger helidon e2e suites. (Note: jaeger hotrod is broken in 1.42.0)

### CODE REVIEW

File: platform-operator/thirdparty/charts/jaegertracing/jaeger-operator/COMPATIBILITY.md [hunk 1]
```diff
@@ -2,6 +2,12 @@ The following table shows the compatibility of `Jaeger Operator helm chart` with
 
 | Chart version             | Jaeger Operator | Kubernetes      | Strimzi Operator   | Cert-Manager |
 |---------------------------|-----------------|-----------------|--------------------|--------------|
+| 2.41.0                    | v1.42.x         | v1.19 to v1.24  | v0.23              | v1.6.1+      |
+| (Missing)                 |                 | v1.19 to v1.23  | v0.23              | v1.6.1+      |
```
Comment (kminder) [inline platform-operator/thirdparty/charts/jaegertracing/jaeger-operator/COMPATIBILITY.md:6]: What do the (Missing) and blank values mean?

Comment (pshirodkar-orcl) [reply platform-operator/thirdparty/charts/jaegertracing/jaeger-operator/COMPATIBILITY.md:6]: https://github.com/jaegertracing/helm-charts/blob/jaeger-operator-2.41.0/charts/jaeger-operator/COMPATIBILITY.md

> (Missing) Missing chart version for specified Jaeger Operator version

File: platform-operator/verrazzano-bom.json [hunk 2]
```diff
@@ -597,15 +597,15 @@
...
+      "version": "1.42.0",
       "subcomponents": [
         {
           "repository": "verrazzano",
           "name": "jaeger-operator",
           "images": [
             {
               "image": "jaeger-operator",
-              "tag": "1.37.0-20230209065122-0760c0d3",
+              "tag": "1.42.0-20230327230723-9970d003",
```
Comment (mgianatagh) [inline platform-operator/verrazzano-bom.json:608]: The hash for suffix of this tag is different than the other Jaeger images.  Just checking if that is ok.

Comment (pshirodkar-orcl) [reply platform-operator/verrazzano-bom.json:608]: jaeger-operator image is created by a separate build from source job. Different from jaeger build from source job.

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**